### PR TITLE
Adding Unit Tests for Intersection.py

### PR DESCRIPTION
**Pull request by mhhwang1**
Updated Intersection.py the plane_lines_intersection function that was built for Maddie's use in TstMotionBasedCanting. Also added a unit test for Intersection.py to check for different cases.

### CODE REVIEW

**Comment by e10harvey** [comment]
Thank you, @mhhwang1. @bbean23 will review.

File: opencsp/common/lib/csp/test/test_Intersection.py [hunk 1]
```diff
@@ -108,9 +110,105 @@ def test_converging_reflection(self):
         # Test
         np.testing.assert_array_almost_equal(intersection.intersection_points._data, expected_points.data)
 
+    def test_bad_params_raises_error(self):
```
Comment (bbean23) [inline opencsp/common/lib/csp/test/test_Intersection.py:113]: These tests look great! 🥳

**Review by bbean23 — COMMENTED** [review]
I only have a couple of comments. Please respond and then I'll approve.

**Comment by e10harvey** [comment]
@mhhwang1: Do you have what you need to finish this?

**Comment by mhhwang1** [comment]
@e10harvey yes, I apologize I've been out of town.

File: opencsp/common/lib/geometry/Intersection.py [hunk 2]
```diff
@@ -128,32 +130,61 @@ def plane_intersect_from_ray_trace(
...
+        cls,
         lines: tuple[Pxyz, Vxyz],
         plane: tuple[Pxyz, Uxyz],  # used to be --> plane_point: Pxyz, plane_normal_vector: Uxyz,
         epsilon: float = 1e-6,
         verbose: bool = False,
-    ):
+    ) -> Pxyz:
         """Vectorized plane intersection algorithm
-        place = (plane_point, plane_normal_vector)"""
+        plane = (plane_point, plane_normal_vector)
```
[the remark this answers is not in the document]
Comment (mhhwang1) [reply opencsp/common/lib/geometry/Intersection.py]: changed exactly like your docstring!

File: opencsp/common/lib/geometry/Intersection.py [hunk 3]
```diff
@@ -171,23 +202,23 @@ def _from_ray_trace_vec_maddie(
         intersection_points = Pxyz(intersection_matrix)
 
         # filter out points that miss the plane
-        if verbose:
-            print("filtering out missed vectors")
-        filtered_intersec_points = (
-            intersection_points  # Pxyz.merge(list(filter(lambda vec: not vec.hasnan(),intersection_points)))
-        )
+        # if verbose:
```
[the remark this answers is not in the document]
Comment (mhhwang1) [reply opencsp/common/lib/geometry/Intersection.py]: unnecessary code that was pulled from his previous function that I removed.

File: opencsp/common/lib/geometry/Intersection.py [hunk 4]
```diff
@@ -128,32 +130,61 @@ def plane_intersect_from_ray_trace(
...
         plane_normal_vector = plane_normal_vector.normalize()
         plane_vectorV = plane_normal_vector.data  # column vector
         plane_pointV = plane_point.data  # column vector
 
         # most recent points in light path ensemble
-        if verbose:
-            print("setting up values...")
+        lt.debug("setting up values...")
         P = points.data
         V = directions.data  # current vectors
```
[the remark this answers is not in the document]
Comment (mhhwang1) [reply opencsp/common/lib/geometry/Intersection.py:200]: also removed verbose arguments elsewhere that were commented out.

**Review by bbean23 — APPROVED** [review]
approved!

**Pull request merged**